Updates and polish for Manifest V3 content scripts docs (#1984)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/reference/scripting/index.md

## Usage

You can use the `chrome.scripting` API to inject JavaScript and CSS into
websites. This is similar to what you can do with [content

> scripts][contentscripts], but by using the `chrome.scripting` API, extensions

Reviewer: ```suggestion
scripts][contentscripts]. But by using the [`chrome.scripting` namespace](/docs/extensions/reference/scripting/), extensions
```

main frame of the specified tab.

```js
function getTabId() { ... }

chrome.scripting
  .executeScript({
    target: {tabId: getTabId()},
    files: ['script.js'],

>   })

Reviewer 2: Throughout this CL, I'd recommend using google JS style / clang --js style.  Otherwise, we'll continue to have inconsistent styling.

Author: Ack. Preparing another PR to add some guidance to the dcc handbook about this. I'll also update this PR to follow this recommendation.

main frame of the specified tab.

```js
function getTabId() { ... }

chrome.scripting
    .executeScript({
      target : {tabId : getTabId()},
      files : [ "script.js" ],
    })
    .then(() => console.log("script injected"));
```

To run in all frames of the specified tab, you can set the `allFrames` boolean
to `true`.

```js
function getTabId() { ... }

chrome.scripting
    .executeScript({
      target : {tabId : getTabId(), allFrames : true},
      files : [ "script.js" ],
    })
    .then(() => console.log("script injected in all frames"));
```

You can also inject into specific frames of a tab by specifying individual frame

> IDs. For more information on frame IDs, see the [webNavigation
> API][webnavigation].

Reviewer: ```suggestion
IDs. For more information on frame IDs, see the [`chrome.webNavigation`
API][webnavigation].
```

Reviewer: I'm not sure whether to use API or namespace here.

async function addIframe() {
  const iframe = document.createElement("iframe");
  const loadComplete =
      new Promise(resolve => iframe.addEventListener("load", resolve));
  iframe.src = "https://example.com";
  document.body.appendChild(iframe);
  await loadComplete;
  return iframe.contentWindow.document.title;
}

chrome.scripting
    .executeScript({
      target : {tabId : getTabId(), allFrames : true},
      func : addIframe,
    })
    .then(injectionResults => {
      for (const frameResult of injectionResults) {
        const {frameId, result} = frameResult;
        console.log(`Frame ${frameId} result:`, result);
      }
    });
```

## Examples

### Unregister all dynamic content scripts

The following snippet contains a function that unregisters all dynamic content
scripts the extension has previously registered.

```js
async function unregisterAllDynamicContentScripts() {
  try {
    const scripts = await chrome.scripting.getRegisteredContentScripts();
    const scriptIds = scripts.map(script => script.id);
    return chrome.scripting.unregisterContentScripts(scriptIds);
  } catch (error) {
    const message = [
      "An unexpected error occurred while",

>       "unregistering dynamic content scripts",

Reviewer: ```suggestion
      "unregistering dynamic content scripts.",
```

Reviewer: I normally wouldn't have bothered with this, but you punctuated an error message farther down.

  return iframe.contentWindow.document.title;
}

chrome.scripting
  .executeScript({
    target: {tabId: getTabId(), allFrames: true},
    func: addIframe,
  })
  .then(injectionResults => {
    for (const frameResult of injectionResults) {

>       console.log('Iframe Title: ' + frameResult.result);

Reviewer 3: Since the code from documentation is often copied by beginners, their actual `result` may be an object or an array of objects that will be rendered as a useless `[object Object]`, so it'd be nice if the documentation would use `,` instead of `+` in all examples.

Author: Solid point. I'll update update this PR now to integrate that feedback.

  return iframe.contentWindow.document.title;
}

chrome.scripting
  .executeScript({
    target: {tabId: getTabId(), allFrames: true},
    func: addIframe,
  })
  .then(injectionResults => {
    for (const frameResult of injectionResults) {
      const {frameId, result} = frameResult;
      console.log(`Frame ${frameId} result:`, result);
    }
  });
```

> ## Examples

Reviewer 2: It seems like all the preceding code examples were also examples - what determines when we put one here vs there?

Author: The short answer is recency. I began adding "Examples" headers to other API reference pages around a year ago in an attempt to highlight common patterns and ways that these APIs are used. The existing examples on this page predate that work and while authoring this PR I felt that normalizing the organization of the samples would be too much scope creep for this PR.

Author: I'd like to try to land this PR as is and follow up with additional cleanup.

Reviewer: It looks like 'Usage' is more like 'this API 101' while examples are use case specific. That's a valid distinction, but I think it will be lost on most readers. Labeling it that way might create unreasonable expectations about how much we can provide in this section. I think it's also inevitable that there will be examples where the label they should be under is not clear. At that point the distinction will become meaningless. 

I would simplify and come up with one heading to rule them all.
